src/charm.py

```diff
@@ -227,6 +239,59 @@ def _on_database_storage_detaching(self, _) -> None:
         # Inform other hooks of current status
         self._peers.data[self.unit]["unit-status"] = "removing"
 
+    def _on_update_status(self, _) -> None:
```

Comment:
Is there are reason you handle this via `update-status` and manual recovery functions?`systemd` does this for you and it might be best to allow `systemd` to handle reboots rather than doing this in status checks (because systemd is a proven resilient method and because update status might be configured to wait 5mins or 60mins thus delaying any necessary self healing)

If you're curious how to use `systemd` to add self healing check [this pr](https://github.com/canonical/mongodb-operator/pull/77/files) or [this snipet](https://github.com/MiaAltieri/DPE-Templates/blob/main/self-healing/self_heal_example.py)